### ISSUE TITLE
Control panel doesn't open

### ISSUE THREAD

**Issue by RodBarnes** [geoffreybennett/alsa-scarlett-gui#109]
# `alsa-scarlett-gui` Issue Template

Thank you for taking the time to contribute to the `alsa-scarlett-gui` project. Before you submit your issue, please ensure you have checked the FAQ and provide the necessary information below.

## Confirmation
- [x] I confirm that I have read the [FAQ](https://github.com/geoffreybennett/alsa-scarlett-gui/blob/master/FAQ.md).

## Issue Category
Please select the category that best describes your issue:
- [x] Help Request
- [ ] Bug Report
- [ ] Feature Request

## Environment Details
Please provide the following details about your environment.

### Linux Distribution and Version
- Distribution: LinuxMint
- Version: 21.3

### Kernel Version
- Kernel version: 6.5.0-26-generic

### Kernel Messages
(paste output from `dmesg | grep -A 5 -B 5 -i focusrite` here)
```
[    2.272447] i915 0000:00:02.0: [drm] Protected Xe Path (PXP) protected content support initialized
[    2.280605] usb 3-6: new high-speed USB device number 3 using xhci_hcd
[    2.437694] usb 3-6: New USB device found, idVendor=1235, idProduct=8212, bcdDevice= 6.45
[    2.437702] usb 3-6: New USB device strings: Mfr=1, Product=3, SerialNumber=2
[    2.437706] usb 3-6: Product: Scarlett 4i4 USB
[    2.437708] usb 3-6: Manufacturer: Focusrite
[    2.437710] usb 3-6: SerialNumber: D8QF64K0986C8A
[    2.488270] [drm] Initialized i915 1.6.0 20201103 for 0000:00:02.0 on minor 0
[    2.491235] ACPI: video: Video Device [GFX0] (multi-head: yes  rom: no  post: no)
[    2.492279] input: Video Bus as /devices/LNXSYSTM:00/LNXSYBUS:00/PNP0A08:00/LNXVIDEO:00/input/input3
[    2.534429] fbcon: i915drmfb (fb0) is primary device
--
[    4.832033] input: HDA Intel PCH HDMI/DP,pcm=7 as /devices/pci0000:00/0000:00:1f.3/sound/card0/input12
[    4.832060] input: HDA Intel PCH HDMI/DP,pcm=8 as /devices/pci0000:00/0000:00:1f.3/sound/card0/input13
[    4.832096] input: HDA Intel PCH HDMI/DP,pcm=9 as /devices/pci0000:00/0000:00:1f.3/sound/card0/input14
[    4.878496] iwlwifi 0000:00:14.3: base HW address: 80:38:fb:5f:0a:ff
[    4.924707] iwlwifi 0000:00:14.3 wlo1: renamed from wlan0
[    6.166334] usb 3-6: Focusrite Scarlett Gen 3 Mixer Driver 6.5-v1.0 enabled (pid=0x8212); report any issues to https://github.com/geoffreybennett/scarlett-gen2/issues
[    6.193338] usb 3-6: Firmware version 1605
[    6.197266] usb 3-6: Quirk or no altest; falling back to MIDI 1.0
[    6.197538] usbcore: registered new interface driver snd-usb-audio
[    6.299253] zfs: module license 'CDDL' taints kernel.
[    6.299256] Disabling lock debugging due to kernel taint

```
### Focusrite Interface Series and Model
- Series: Scarlett 3rd gen
- Model: 4i4

### Audio System
- [x] PulseAudio
- [ ] JACK
- [x] PipeWire

## Issue Description
At this point, I'm just trying to run the GUI.  I downloaded the release [alsa-scarlett-gui_0.4.0_amd64.deb](https://github.com/geoffreybennett/alsa-scarlett-gui/releases) and installed it.  It did not produce any errors during install but when I invoke it from the  menu, nothing appears.

I'm willing to try building it (as described [here](https://github.com/geoffreybennett/alsa-scarlett-gui/blob/master/docs/INSTALL.md)) but, given a release is available, wanted to try success via that route, first.

---

Thank you for helping improve `alsa-scarlett-gui`!

**Comment by geoffreybennett** [comment]
Hi @RodBarnes,

Could you try running it from the command-line: `alsa-scarlett-gui` and paste the output here please.

Thanks,
Geoffrey.

**Comment by RodBarnes** [comment]
Well, now I don't know what to think.  I just reinstalled it a third time to try this and it is now working.  I installed the GUI the first time and it would not appear but this was before I had installed the driver.  So, after learning what I needed to do there, I uninstalled the GUI, downloaded and built the driver, rebooted, and confirmed the driver via the `dmesg` command.  I then installed the GUI again and it still would not appear.  I just now installed it again, and ran it from the command line and it appeared (though, it did display the message at bottom).  I then closed it and invoked it from the menu, and it again appeared.  So, not sure what was happening.

Maybe it took running it from the command-line once, which I had not done before?

I tried the controls and they all seem to work.  So, sorry for the false alarm I guess... But, I'll be making a donation for all your work on this.  Astounding.

The message I saw when running it from the command line:
`Firmware directory /usr/lib/firmware/scarlett2 does not exist`

**Comment by geoffreybennett** [comment]
@RodBarnes I found the issue; the deb package is missing the dependency on Gtk4. I'll figure out how to add that to the deb package and make a new release. I guess at some point in trying things you installed the libgtk-4 package.

The "Firmware directory...does not exist" message will go away if you `sudo mkdir /usr/lib/firmware/scarlett2`. You could also install the firmware from https://github.com/geoffreybennett/scarlett2-firmware but you are already on the latest firmware, so that won't do much.

Thank you for your support!

**Comment by RodBarnes** [comment]
> I guess at some point in trying things you installed the libgtk-4 package.

That explains it. When I couldn't get it to work before, I went to the build page and began installing the prerequisites:
`sudo apt -y install git make gcc libgtk-4-dev libasound2-dev libssl-dev`

But, then I decided to put that on hold and first check why the downloaded release didn't work.  I did the message in that page about requiring GTK4 libraries.  Just didn't connect that adding them might fix the release install. :)

> Thank you for your support!
Certainly. I'm very impressed with what you've done here.
